### ISSUE TITLE
Cache any value you read from solana which is not controlled by neon alone.

### ISSUE THREAD

**Issue by anton-lisanin** [neonevm/neon-evm#296]
- you can change solana chain state while execution of eth is running, and some of it will reflect inside the evm. This is a problem, as contract devs will usually assume nothing can change while their program executes. This can cause bugs in contracts which would be save on ethereum. Specifically: the values of current block, block_time and spl token balances can change:
    - the erc20-wrapper-syscall can provide decimals, total_supply and `balance`s of token accounts, all of which are changeable between multiple the transactions of one neon iterative executions.
    - basically everything with apply_to_solana_account is affected.
    - a similar bug also affects the re-initialization of the ProgramAccountStorage when execution resumes. SPL balances are read fresh. This is an issue if (erc20) tokens are traded without neon on native solana, though balance can only increase (because account is blocked, so neon won't touch it).
    - to fix this, you'll need to cache any value you read from solana which is not controlled by neon alone.
